Add Touchscreen Controls to HUD

jbourqueendless opened the pull request:
- Introduced touchscreen controls for the game by adding TouchScreenButton nodes to the HUD.
- Controls include buttons for movement (left and right) and a jump action.
- Buttons are positioned as part of the HUD layout and are only visible on touchscreen devices.
- Tested functionality on desktop (using "Emulate Touch From Mouse") and mobile device.

cassidyjames commented:
Ooh, neat idea! It looks like your PR added a bunch of unused assets and the block-code addon as well. 😅 I can help you clean it up if you'd like, because I think this is a nice idea (and the clean-up process will be… informative? lol)

(@endlessm/endless-developers: we should probably .gitignore `addons/`?)

jbourqueendless commented:
OK I gotcha. So between the block coding and the images that came with the interface button asset pack what should be removed? All of the block coding plug in and any and all images not directly used or?

cassidyjames commented:
@jbourqueendless full disclaimer: I haven't downloaded this branch to play with it yet. But generally, yeah, I would remove the block coding plugin, and then maybe instead of keeping all of the assets in their original locations/names (like `assets/Touchscreen Controls/kenney_onscreen-controls/Sprites/flat-dark/flatDark05.png`), I would only import the specific assets you're going to use, and name them something simpler (like `assets/controls/05.png`).

manuq commented:
@jbourqueendless yes, as @cassidyjames said, only `git add` the needed assets for the 3 buttons. Ideally with the same file name format as the existing assets, something like:
```
assets/hud/touchscreen-button-left.png
assets/hud/touchscreen-button-right.png 
assets/hud/touchscreen-button-a.png 
```

jbourqueendless commented:
Ok I have removed all but a small selection of control button pngs from the asset pack. I also reorganized them and renamed them (which I am adding here because I forgot to mention that in the commit message :-/ ). Block coding plugin and files are also removed.

wjt commented on project.godot:
```diff
@@ -31,6 +31,8 @@ window/stretch/mode="canvas_items"
 [global_group]
 
 players=""
+switch=""
+animator=""
```
I think this (and the addition of `switch.tscn`) are also not meant to be part of this PR.

wjt commented:
> (@endlessm/endless-developers: we should probably .gitignore `addons/`?)

In this one we don't use any addons right now, but in moddable pong we do: https://github.com/endlessm/moddable-pong/tree/main/addons/DampedOscillator. I don't think ignoring it would be right.

wjt commented on assets/controls/flatDarkB.png.import:
```diff
@@ -0,0 +1,34 @@
+[remap]
```
Strictly speaking the B, Up and Down assets in each colour scheme are not needed (but to me it seems harmless to have them. You never know what new features might need them later).

cassidyjames replied:
Justin mentioned wanted having an alternate option there as part of the modability, so I think it can make sense to keep (similar to how we have multiple players sprites).

wjt replied:
I pushed a patch to use the inverted assets while the buttons are pressed.

wjt commented on switch.tscn:
```diff
@@ -0,0 +1,102 @@
+
+[ext_resource type="Texture2D" path="res://assets/switch.png.png" id="1_gec7q"]
+[ext_resource type="Script" path="res://addons/block_code/block_code_node/block_code.gd" id="2_eobrb"]
+[ext_resource type="Script" path="res://addons/block_code/serialization/block_serialization_tree.gd" id="3_o6qre"]
```
This scene won't load because you (correctly) removed the block code plugin from this PR – I think this scene should be removed from this PR too.

wjt commented:
I've tested this on my laptop touchscreen: it looks great. With the stray changes removed, I'd say it's ready to (squash and) merge.

@jbourqueendless would you like me/us to take care of the stray files and/or squashing? Or do you want to do either/both?

jbourqueendless commented:
That's ok I will remove the scene file and squash all these little commits.

wjt reviewed (approved):
Very nice!
